src/imperator/data/setup_main.rs

```diff
@@ -0,0 +1,295 @@
+            }
+            
+            capital = 1
+            pantheon = {
+            
+            { deity = 1 }
+            { deity = 2 } 
+            { deity = 6 } 
+            { deity = 4 }
+            
+            }
+            is_antagonist = yes
+        
+            treasures = { 201 61 39 }
+        
+            own_control_core =  {
+                1 2 3 4 5 6 7 8 15 16 18 19 20 24 25 26 27 31 37 40 36 39 50
+            }
+            
+            <law_name> = yes
+            #professional_soldiers = yes
+            #organized_recruitment = yes
+        }
+    */
+
+    let mut vd = Validator::new(block, data);
+
+    vd.field_validated_block("countries", |block, data| {
+        let mut vd = Validator::new(block, data);
+
+        vd.validated_blocks(|block, data| {
+            let mut vd = Validator::new(block, data);
+
+            vd.field_item("government", Item::Government);
+            vd.field_item("diplomatic_stance", Item::DiplomaticStance);
+            vd.field_item("religion", Item::Religion);
+            vd.field_item("culture", Item::Culture);
+
+            vd.field_integer("family");
+            vd.field_integer("capital");
+            vd.field_bool("is_antagonist");
+
+            vd.field_validated_block("treasures", |block, data| {
+                let mut vd = Validator::new(block, data);
+                vd.req_tokens_integers_at_least(1);
+            )};
+            vd.field_validated_block("own_control_core", |block, data| {
+                let mut vd = Validator::new(block, data);
+                vd.req_tokens_integers_at_least(1);
+            });
+
+            // TODO - laws, deities, and technology
+        });
+    });
+}
+
+fn validate_trade(block: &Block, data: &Everything) {
+    let mut vd = Validator::new(block, data);
+
+    vd.multi_field_validated_block("route", |block, data| {
+        let mut vd = Validator::new(block, data);
+
+        vd.field_integer("from");
+        vd.field_integer("to");
+        vd.field_item("trade_goods", Item::TradeGood);
+    });
+}
+
+fn validate_great_works(block: &Block, data: &Everything) {
+    /*
+        Example:
+        1={
+            ancient_wonder = yes
+            key="temple_of_jupiter"
+            great_work_category="building"
+            great_work_state=great_work_state_completed
+            finished_date=450.10.1
+
+            great_work_name={
+                name="wonder_jupiter_temple"
+            }
+            great_work_components={
+                {
+                    great_work_module="wonder_temple_of_jupiter"
+                }
+            }
+
+            great_work_effect_selections={
+                {
+                    great_work_effect="gw_effect_omen_doctrine"
+                    great_work_effect_tier = "gw_effect_tier_4"
+                }
+            }
+        }
+
+        How to match the loose brackets in the great_work_components and great_work_effect_selections blocks???
```

Comment:
Those are known as "loose blocks" and there are some methods for iterating over them. Might be as simple as validate_blocks, but my memory is spotty.